protocols/index/reports/SECURITY.md

```diff
@@ -0,0 +1,151 @@
+- **This invariant currently holds as-is.**
+
+
+#### 3. Linked list ordering by score must be preserved at all times (Highest -> Lowest).
+
+a. Inserting Locators preserves the ordering of the list.
+By induction:
+- Case where the list is empty:
+    - There are no locators in the list and the list is merely the HEAD that points to itself (this is setup in the constructor).
+    - The invariant is vacuously true for the base case
+- Given a new Locator, we call setLocator
+    - Assuming the new locator is a valid Locator.
+        - set by the owner
+        - has a user who has not previously set a Locator
+    - The location to insert the new locator in is found using the function findPosition(score)
+        - If its score is 0, the location for the new locator is the end of the list
+        - Otherwise findPosition looks at the list of locators in decreasing score value, and returns the first score that is lower than the new score.
+        - If there is an equal score, the new Locator is placed after the previous same-scored Locator.
+    - This is guaranteed to happen, as we know the new locator has a score of more than 0, and HEAD has a score of 0, meaning if the new locator is the new tail of the list the HEAD will be returned.
+    - The Locator is then inserted in its rightful position, and the list is still in descending order of score.
+- **This invariant currently holds as-is.**
+b. Removing locators preserves the ordering of the list
+- Given a list starting with n > 0 locators in it, one locator, x, is removed by calling unsetLocator
+unsetIntent first links together the element before and after x in the list so that they point to each other instead of to x
+- The data stored in x’s pointers is then deleted, so that if it is looked up in future it is clear that this locator does not exist
+- The edge case is when one of the ‘locators’ before or after x in the list is actually the head of the list (meaning x was first and/or last in the list).
+- In this case, the head is simply linked to the new first/last element in the list, and the ordering is preserved.
+- **This invariant currently holds as-is.**
+
+#### 4. Each address can have either 0 or 1 locators on the index at any time - and therefore calling setLocator() twice does not disrupt the ordering of the locator linked list.
+
+- When the contract is created, the list is initialised as empty, and therefore no Locators exist for any user addresses.
+- When a new Locator is being set, if the address already has a Locator, the transaction reverts with LOCATOR_ALREADY_SET ensuring addresses cannot have more than 1 Locator on the index.
+- **This invariant currently holds as-is.**
+
+#### 5. Calling unsetLocator() twice does not disrupt ordering of the locator linked list.
+
+- At the beginning of unsetLocator, the function checks whether the specified address actually has an address. If they do not, the function returns false and stopped executing
```

Comment:
change to "At the beginning of unsetLocator, the function checks whether the specified address actually has an address. If they do not, the function returns false and stops executing"